src/Components/PaperEditor/ProgrammeScript/index.js

```diff
@@ -1,238 +1,269 @@
-      resetPreview: false,
-      width: 150,
-      // demo content
-      playlist: [
-        // start - is relative to timeline
-        // duration - of clip in playlist
-        // sourceStart - time from start of clip in playlist
-        // { type:'video', start:0, sourceStart: 30, duration:10, src:'https://download.ted.com/talks/MorganVague_2018X.mp4' },
-        // { type:'video', start:10, sourceStart: 40, duration:10, src:'https://download.ted.com/talks/IvanPoupyrev_2019.mp4' },
-        // { type:'video', start:20, sourceStart: 50, duration:10, src:'https://download.ted.com/talks/KateDarling_2018S-950k.mp4' },
-      ]
-    };
-  }
 
-  updateVideoContextWidth = () => {
-    this.setState({
-      width: this.card.offsetWidth - 10
-    });
-  };
+import PreviewCanvas from '@bbc/digital-paper-edit-storybook/PreviewCanvas';
+import ProgrammeElements from '@bbc/digital-paper-edit-storybook/ProgrammeElements';
 
-  getTranscript = transcriptId => {
-    return this.props.transcripts.find(tr => tr.id === transcriptId);
-  };
+import Collection from '../../Firebase/Collection';
+import { withAuthorization } from '../../Session';
 
-  getPlayList = () => {
-    let startTime = 0;
-
-    return this.state.programmeScript.elements
-      .filter(element => element.type === 'paper-cut')
-      .map(element => {
-        // TODO: handle audio only type (eg for radio), get from transcript json?
-        const transcript = this.getTranscript(element.transcriptId);
-        const result = {
-          type: 'video',
-          start: startTime,
-          sourceStart: element.start,
-          duration: element.end - element.start,
-          src: transcript.url
-        };
+import ExportDropdown from './ExportDropdown/index';
+import ElementsDropdown from './ElementsDropdown/index';
+import getDataFromUserWordsSelection from './get-data-from-user-selection';
+import { divideWordsSelectionsIntoParagraphs, isOneParagraph } from './divide-words-selections-into-paragraphs';
 
-        startTime += result.duration;
+import PropTypes from 'prop-types';
 
-        return result;
+const ProgrammeScript = props => {
+  const transcripts = props.transcripts;
+  const papereditsId = props.match.params.papereditId;
+  const projectId = props.match.params.projectId;
+
+  const [ elements, setElements ] = useState();
+  const [ title, setTitle ] = useState('');
+  const [ resetPreview, setResetPreview ] = useState(false);
+
+  const SortableList = SortableContainer(({ children }) =>
+    <ul style={ { listStyle: 'none', padding: '0px' } }>
+      {children}
+    </ul>
+  );
+
+  // Video Context Preview
+  const [ width, setWidth ] = useState(150);
+  const [ playlist, setPlaylist ] = useState();
+  const previewCardRef = useRef(null);
+
+  const PaperEditsCollection = new Collection(
+    props.firebase,
+    `/projects/${ projectId }/paperedits`
+  );
+
+  const handleSaveProgrammeScript = async () => {
+    console.log('Saving...');
```

Comment:
Did we want to keep this until we actually have some indicator for progress of save?